main.c

```diff
@@ -85,13 +85,18 @@ main(int argc, char **argv)
     case UNKNOWN:
     case RUN:	    /* Run the specified (or all the discovered) tests */
 	/* Set the output format */
-	if (output_format)
+	if (output_formats)
 	{
-	    if (!np_set_output_format(runner, output_format))
-	    {
-		fprintf(stderr, "np: unknown output format '%s'\n", output_format);
-		exit(1);
-	    }
+		char *format = strtok (output_formats, ",");
```

Comment:
I can understand your concerns. I'll convert this code over to use tok_t and submit again in a few days.